src/extensions/ExtensionManager.as

```diff
@@ -212,6 +212,10 @@ public class ExtensionManager {
 					// 'r' is reporter, 'R' is requester, and 'rR' is a reporter forced to act as a requester.
 					spec[0] = 'rR';
 				}
+				if(spec[0] == 'h') {
+					// same as above, but force to use async hat blocks
+					spec[0] = 'hH';
+				}
```

Comment:
@cpseager @bsb20 I think that's an argument to either:
- make sure that `'rR'` gets saved as `'r'` in an SB2, or
- use another way to decide to use this behavior instead (obviating the need for the `'r'` -> `'rR'` swap in the first place).

I don't think I was aware of this problem. I've filed a new issue to track it, since I didn't find one in the editor repositories: https://github.com/LLK/scratch-flash/issues/1051